examples/platforms/nrf52840/usb-cdc-uart.c

```diff
@@ -363,25 +378,26 @@ otError otPlatUartDisable(void)
 
 otError otPlatUartSend(const uint8_t *aBuf, uint16_t aBufLength)
 {
+    otError error = OT_ERROR_NONE;
+
+    otEXPECT_ACTION(sUsbState.mTransferInProgress == false, error = OT_ERROR_BUSY);
+    otEXPECT_ACTION(sUsbState.mTxBuffer == NULL, error = OT_ERROR_BUSY);
+
     if (!isPortOpened())
```

Comment:
Shouldn't `sUsbState.mOpenTimestamp` be checked here as well? If there were no queued data and we call `otPlatUartSend` just after the port is opened we will send it without waiting the `USB_HOST_UART_CONFIG_DELAY_MS `.

I can see though, that in case we check for this in `otPlatUartSend`, the time checking logic should be protected against timer overflow, something like:
```
bool hasPortOpenDelayPassed(void)
{
    int32_t timeDiff = otPlatAlarmMilliGetNow() - sUsbState.mOpenTimestamp;

    return (timeDiff < 0) || (timeDiff > USB_HOST_UART_CONFIG_DELAY_MS);
}
```

Reply:
I wasn't too worried about that case because it is effectively guaranteed not to happen with the way the NCP code is currently written for the specific scenario we are trying to prevent (failing to receive the reset reason).

Adding the check would end up causing a periodic delay of 10ms every few weeks (due to roll-over), but that seems so insignificant I can't really consider it a concern.

I'll go ahead and address and test that today so that we can get this merged in ASAP.